src/OpenTelemetry.Instrumentation.Runtime/MeterProviderBuilderExtensions.cs

```diff
@@ -33,7 +33,7 @@ public static class MeterProviderBuilderExtensions
     /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
     public static MeterProviderBuilder AddRuntimeInstrumentation(
         this MeterProviderBuilder builder,
-        Action<RuntimeInstrumentationOptions> configure = null)
+        Action<RuntimeInstrumentationOptions>? configure = null)
```

Comment:
Related conversation here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3653#discussion_r969915942. I think the direction we'd like to head is to remove optional parameters.

```suggestion
        Action<RuntimeInstrumentationOptions> configure)
```

@CodeBlanch seem correct?

Reply:
@CodeBlanch just reminded me that you'd need to also add a new overload to make this work

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder)
```

Additionally, in the opentelemetry-dotnet repo an named option version of these extensions has been introduced as well. Could be a separate PR.

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(
        this MeterProviderBuilder builder,
        string? name,
        Action<RuntimeInstrumentationOptions> configure)
```

Reply:
@alanwest, what is your preference about nullability for cofnigure parameter

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder)
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder, Action<RuntimeInstrumentationOptions> configure)
```

or

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder)
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder, Action<RuntimeInstrumentationOptions>? configure)
```

I would keep name attribute as separate PR/issue.

Reply:
The first set is the only one that would work. The second option results in two ambiguous methods.

> I would keep name attribute as separate PR/issue.
👍

Reply:
Fixed in dff3ace08a4df45a47155a5a57a9fdc3ec90b974.